irohad/ametsuchi/impl/soci_utils.hpp

```diff
@@ -131,15 +131,16 @@ namespace iroha {
 
     template <typename C, typename T, typename F>
     auto flatMapValues(T &t, F &&f) {
-      return t | [&](auto &st) -> boost::optional<C> {
-        return boost::copy_range<C>(
-            st | boost::adaptors::transformed([&](auto &t) {
-              return apply(t, std::forward<F>(f));
-            })
-            | boost::adaptors::filtered(
-                  [](auto &r) { return static_cast<bool>(r); })
-            | boost::adaptors::transformed([](auto &&r) { return *r; }));
-      };
+      C map_result;
+      if (t) {
+        for (auto &inp_el : *t) {
+          apply(inp_el, std::forward<F>(f)) |
+              [&map_result](auto &&transformed_el) {
+                map_result.emplace_back(std::move(transformed_el));
+              };
+        }
+      }
+      return map_result;
```

Comment:
The logic is changed here: previously it returned `boost::none` if `t` was `boost::none`. Please preserve the previous logic without using `boost::adaptors`.